python/xir/program.py

```diff
@@ -430,3 +512,18 @@ def serialize(self, minimize: bool = False) -> str:
         if minimize:
             return strip(res_script)
         return res_script
+
+    @staticmethod
+    def _validate_version(version: str) -> None:
+        """Validates the given version number.
+
+        Raises:
+            TypeError: If the version number is not a string.
+            ValueError: If the version number is not a semantic version.
+        """
+        if not isinstance(version, str):
+            raise TypeError(f"Version '{version}' must be a string.")
+
+        valid_match = re.fullmatch(r"\d+\.\d+\.\d+", version)
+        if valid_match is None or valid_match.string != version:
```

Comment:
```suggestion
        if valid_match is None:
```
I guess this can be removed now, since `fullmatch` should only return either `None` or the full version string anyway.